README.md

```diff
@@ -61,3 +61,40 @@ As mentioned above, package specifications are versioned. Versions follow the [s
 ## Version Compatibility between Packages and Specifications
 
 A package specifying its `format_version` as `x.y.z` must be valid against specifications in the semantic version range `[x.y.z, X.0.0)`, where `X = x + 1`.
+
+
+# What is an Elastic Package?
+
+An Elastic Package is a collection of assets for the Elastic Stack. In addition, it contains manifest files which contain additional information about the package. The exact content and structure of a package are described by the package spec. In the following is a high level overview of a package.
+
+A package is downloaded as a .zip file for the package-registry by Fleet inside Kibana. The zip file is then unpacked and each asset is installed into the related API.
+
+## Asset organisation
+
+In general, the assets are organised by {stack-component}/{asset-type}. For example assets for Elasticsearch ingest pipelines are in the directory `elasticsearch/ingest-pipeline`. The same logic applies to all Elasticsearch, Kibana and Elastic Agent assets.
+
+There is a special directory `data_stream`. All assets inside the data_stream directory must follow the [Data Stream naming scheme](https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme). `data_stream` can contain multiple directory, each with the name of the dataset (or postfix). Inside this directory, the same structure as before for {stack-component}/{asset-type} applies. The difference is that for all these assets, Fleet during installation enforces naming rules related to the [Data Stream naming scheme](https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme). All assets in this directory belong directly or indirectly to data streams.
```

Comment:
```suggestion
There is a special folder `data_stream`. All assets inside the `data_stream` folder must follow the [Data Stream naming scheme](https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme). `data_stream` can contain multiple folders, each with the name of the dataset (or postfix). Inside this folder, the same structure as before for {stack-component}/{asset-type} applies. The difference is that for all these assets, Fleet during installation enforces naming rules related to the [Data Stream naming scheme](https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme). All assets in this folder belong directly or indirectly to data streams.
```

Reply:
What is a dataset? 😄 

If we are going to introduce new terminology (relative to the rest of the documentation here) and especially if that terminology sounds similar but isn't quite the same as other terminology (dataset vs. data stream), we need to explain what this new terminology is.

Otherwise it's going to setup readers for confusion. Some will assume dataset is the same as data stream. Others will just be left confused and start questioning their understanding of data streams, etc.

Reply:
I linked to the data stream naming scheme that should explain it. But nobody is going to jump there to read up on it first. At the same time I do not want to dive to deep into this overview. Any ideas for a way out here?

Reply:
Modified it slightly to leave out dataset.